core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java

```diff
@@ -376,5 +450,201 @@ public Pair<Integer, TimeUnit> refresh(RESTClient client) {
 
       return null;
     }
+
+    private OAuthTokenResponse refreshCurrentToken(RESTClient client) {
+      if (isExpired(expiresAtMillis, System.currentTimeMillis())) {
+        // the token has already expired, attempt to refresh using the credential
+        return refreshExpiredToken(client);
+      } else {
+        // attempt a normal refresh
+        return refreshToken(client, headers(), token, tokenType, OAuth2Properties.CATALOG_SCOPE);
+      }
+    }
+
+    static boolean isExpired(Long expiresAtMillis, long now) {
```

Comment:
The reason why I didn't add a method like this is that it isn't correct when `expiresAtMillis` is `null`. It returns false, but the token may still be expired. I think it's misleading to have a method with this name, so I think we should revert the change to add this.

Reply:
thanks for bringing this up. I'll look into that as part of https://github.com/apache/iceberg/pull/6562